src/tools/pal2img.cpp

```diff
@@ -0,0 +1,76 @@
+/***************************************************************************
+ *   fheroes2: https://github.com/ihhub/fheroes2                           *
+ *   Copyright (C) 2023                                                    *
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ *   This program is distributed in the hope that it will be useful,       *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU General Public License for more details.                          *
+ *                                                                         *
+ *   You should have received a copy of the GNU General Public License     *
+ *   along with this program; if not, write to the                         *
+ *   Free Software Foundation, Inc.,                                       *
+ *   59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.             *
+ ***************************************************************************/
+
+#include <cstdint>
+#include <cstdlib>
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include "image.h"
+#include "image_palette.h"
+#include "image_tool.h"
+#include "serialize.h"
+#include "system.h"
+
+int main( int argc, char ** argv )
+{
+    if ( argc != 3 ) {
+        std::string baseName = System::GetBasename( argv[0] );
+
+        std::cerr << baseName << " generates an image with colors based on a provided palette file." << std::endl
+                  << "Syntax: " << baseName << " palette_file.pal output.bmp" << std::endl;
```

Comment:
This tool may (and will, if built with `FHEROES2_WITH_IMAGE` and I pass the `output.png` as the second parameter) also generate PNG files. You can add this to the description, for example:

```suggestion
                  << "Syntax: " << baseName << " palette_file.pal output.[bmp|png]" << std::endl;
```

or you can not add it - it's up to you.